src/core/common/logging.cpp

```diff
@@ -281,14 +281,15 @@ void otDump(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aId, const
     size_t                        idLen = strlen(aId);
     ot::String<kStringLineLength> string;
 
-    for (size_t i = 0; i < (kWidth - idLen) / 2 - 5; i++)
+    for (size_t i = 0; i < (kWidth - idLen) / 2 - 9; i++)
     {
         string.Append("=");
     }
 
-    string.Append("[%s len=%03u]", aId, static_cast<unsigned>(aLength));
+    string.Append("[%s seqno=%03u len=%03u]", aId, static_cast<const uint8_t *>(aBuf)[2],
```

Comment:
Thanks for submitting this. 

I am not sure about  adding `seqno=%03` here in `otDump()` log line.

Note that `otDump()` is used by all `otDump{Level}{Region}()` functions/macros and should treat its input as generic data blob (it is not necessary a MAC frame). For example it is used  `commissioner.cpp` or `joiner.cpp` where it dumps received `Coap::Message` (in `otDumpCertMeshCoP`).

------

If you want to add `seqno` for the mac frame log dumps, my suggestion is that instead we add a specific `DumpFrame()` method in `Mac` or `SubMac` which then formats  its title string (`aId` passed to `otDump()`)  to include "seqno=%03" in it before calling the common logging functions.

Reply:
I could also add e.g.
```
 #define otDumpDebgMacFrame(aId, aBuf, aLength)
 ```
 to src/core/common/logging.hpp, is that ok as well?